src/Generating/FinishGen.h

```diff
@@ -70,6 +70,71 @@ class cFinishGenNetherClumpFoliage :
 
 
 
+class cFinishGenClumpTopBlock :
+	public cFinishGen
+{
+public:
+	// Contains the meta, type and weight for a clump block
+	struct BlockInfo
```

Comment:
Perhaps choose a different name than `Block`, to avoid confusion. How about `FoliageInfo`?